source/features/restore-file.tsx

```diff
@@ -53,27 +62,51 @@ async function restoreFile(progress: (message: string) => void, menuItem: Elemen
 		throw new Error('Restore failed: File too big');
 	}
 
-	let {pathname} = menuItem.previousElementSibling as HTMLAnchorElement;
-	// Check if file was deleted by PR
+	// Attempt to get details from the `Edit` button in the menu
+	const {pathname: blobOrEditPath} = menuItem.previousElementSibling as HTMLAnchorElement;
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars -- Error comes up even though route and pathToFile are prefixed with a _
+	let [owner, repo, _route, branch, ..._pathToFile] = blobOrEditPath.startsWith('/')
+		? blobOrEditPath.slice(1).split('/') : blobOrEditPath.split('/');
+
 	if (menuItem.closest('[data-file-deleted="true"]')) {
+		// The file was deleted by the PR, restore it from the base commit
 		progress('Undeleting…');
-		const [nameWithOwner, headBranch] = select('.head-ref')!.title.split(':');
-		pathname = `/${nameWithOwner}/new/${headBranch}?filename=${filePath}`;
+		// The `Edit` button doesn't exist if the file is deleted
+		const [nameWithOwner, prBranch] = select('.head-ref')!.title.split(':');
+		owner = nameWithOwner.split('/')[0];
+		repo = nameWithOwner.split('/')[1];
+		branch = prBranch;
```

Comment:
I think this is unnecessary changes since owner and repo are being merged again later. Wasn't `nameWithOwner` already exactly that?

Reply:
Also you can move this line outside the `if` and avoid extracting the same information from `blobOrEditPath`

```js
const [nameWithOwner, prBranch] = select('.head-ref')!.title.split(':');
```

Reply:
Ahh good point, will do

Reply:
Done!